x/collection/keeper/nft.go

```diff
@@ -4,15 +4,31 @@ import (
 	gogotypes "github.com/gogo/protobuf/types"
 
 	sdk "github.com/line/lbm-sdk/types"
-	sdkerrors "github.com/line/lbm-sdk/types/errors"
 	"github.com/line/lbm-sdk/x/collection"
 )
 
+func legacyNFTNotFoundError(k Keeper, ctx sdk.Context, contractID string, tokenID string) error {
+	// legacy
+	if err2 := collection.ValidateLegacyNFTID(tokenID); err2 == nil {
+		return collection.ErrTokenNotExist.Wrap(tokenID)
+	}
+
+	if err2 := collection.ValidateFTID(tokenID); err2 != nil {
+		return collection.ErrTokenNotExist.Wrap(tokenID)
+	}
+	classID := collection.SplitTokenID(tokenID)
+	if _, err2 := k.GetTokenClass(ctx, contractID, classID); err2 != nil {
+		return collection.ErrTokenNotExist.Wrap(tokenID)
+	}
+
+	return collection.ErrTokenNotNFT.Wrap(tokenID)
```

Comment:
I don't disagree with your opinion, so I may commit your suggestion. Just out of curiosity, what's the expected optimization?

Reply:
it's very trivial (register jump). Go's compiler may optimize this automatically but I just suggested this because the conditions are orthogonal. So you can ignore it.

Reply:
As Jaden said, Go's compiler may optimize.
In my opinion, the previous one(single if error handling) looks betters because the code follows go's error-handling convention. 
I don't need to memorize the previous error check condition with previous one(`if error check`), but with new code(`if, else if, else if`) I would try to memorize the previous one which is unnecessary though. 
When I add other error check logic, I'll definitely look up all `if, else if, else if` which is not good.

Reply:
> In my opinion, the previous one(single if error handling) looks better because the code follows go's error-handling convention.

It's the reason behind this logic. Thanks for the clarification.

> As Jaden said, Go's compiler may optimize.

I don't see any practical optimization point. It would be almost same with or without the `if else` statements. Let the machines do what they do well, and we should concentrate on the human's job, like the readability. If we can come up with some trivial optimization, they can too.

Reply:
I agree with going for readability :)